### PR TITLE
Feature territory

### LINKED ISSUE #21: Implement territory feature

### PR DESCRIPTION
**Pull request by wilsonjefferson**
This branch is refereed to `Board` class, you can find some methods to find _regions_ and _territories_. This branch was forked by Exception branch.

### CODE REVIEW

**Review by stefanosimonetto — APPROVED** [review]

File: src/main/java/sdmExam/Board.java [hunk 1]
```diff
@@ -76,7 +76,88 @@ public boolean existsDiagonallyAdjacentWithStone(Intersection intersection, Ston
                 .anyMatch(diagonalIntersection -> diagonalIntersection.hasStone(stone));
     }
 
+    public ArrayList<ArrayList<Intersection>> findTerritories() {
+        ArrayList<ArrayList<Intersection>> territories = new ArrayList<>();
+        ArrayList<Intersection> territory = new ArrayList<>();
+
+        for(int col = 1; col <= Board.BOARD_SIZE; col++){
+            Intersection intersection = new Intersection(Position.in(3, col), Stone.NONE);
```
Comment (demirbilek95) [inline src/main/java/sdmExam/Board.java]: Why do we have 3 as hard coded here? (In Position.in(3, col)). Are we looking for territories only in row 3?

File: src/main/java/sdmExam/Board.java [hunk 2]
```diff
@@ -76,7 +76,88 @@ public boolean existsDiagonallyAdjacentWithStone(Intersection intersection, Ston
...
+
+    public ArrayList<ArrayList<Intersection>> findRegions() {
+        ArrayList<ArrayList<Intersection>> regions = new ArrayList<>();
+
+        int h = intersections.size();
+        if (h == 0)
+            return regions;
+
+        for (int i = 0; i < h; i++) {
+            if (!intersections.get(i).isOccupied() && findIntersectionInRegions(regions, intersections.get(i))) {
```
Comment (demirbilek95) [inline src/main/java/sdmExam/Board.java:125]: intersections.get(i).isOccupied() is tending to code smell, we may need to find another way to perform

**Review by demirbilek95 — CHANGES_REQUESTED** [review]
I think we need some refactoring here, we can talk the details with call

**Comment by eferos93** [comment]
Closed until the feature is not implemented properly

**Pull request closed without merging**